hw/ip/otbn/dv/rig/rig/gens/bad_ispr.py

```diff
@@ -93,5 +94,10 @@ def _gen(self,
             bad_addr = model.pick_bad_addr(mem_type)
             assert bad_addr is not None
             prog_insn.lsu_info = mem_type, bad_addr
+            for i, each in enumerate(prog_insn.insn.operands):
+                if isinstance(each.op_type, ImmOperandType):
+                    imm_idx = i
+            assert imm_idx is not None
```

Comment:
This needs a line `imm_idx = None` just before the for loop. Python's lax scoping rules mean that the existing code will work unless we never see an immediate operand but, in that case, we'll get an unbound variable error for `imm_idx` (rather than it existing, but having value `None`).

Also, could you add an assertion just before assigning `imm_idx` that says `assert imm_idx is None`. That will check that we don't have two different immediate operands.

Reply:
Added the required lines